src/main/java/io/github/a5h73y/parkour/gui/JoinCourses.java

```diff
@@ -49,7 +49,11 @@ public GuiElementGroup getGroupContent(InventoryGui parent, Player player) {
 
 							// the item description
 							TranslationUtils.getValueTranslation("GUI.JoinCourses.Description",
-									course, false)
+									course, false),
+							TranslationUtils.getValueTranslation("GUI.JoinCourses.Players",
+									String.valueOf(Parkour.getInstance().getPlayerManager().getNumberOfPlayersOnCourse(course)), false),
+							TranslationUtils.getValueTranslation("GUI.JoinCourses.Checkpoints",
+									String.valueOf(Parkour.getInstance().getCourseManager().getCourse(course).getNumberOfCheckpoints()), false)
```

Comment:
Instead of looking up the entire course entity, could you instead use CourseInfo.getCheckpointAmount(course) as it could be considerably faster.

Reply:
Thanks - yes good point, forgot that existed